spec/intl.html

```diff
@@ -238,8 +241,12 @@ <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
         1. <del>Set _dateTimeFormat_.[[HourCycle]] to _hc_.</del>
         1. Let _timeZone_ be ? Get(_options_, *"timeZone"*).
         1. If _timeZone_ is *undefined*, then
-          1. Set _timeZone_ to DefaultTimeZone().
+          1. <ins>If _toLocaleStringTimeZone_ is present, then</ins>
```

Comment:
I'll defer to Shane as to whether this is actually a concern, but my guess is that an implementation would want to avoid entering the DTF constructor in the first place, if the ZonedDateTime could already determine that the formatting operation was going to fail because we have a weird custom time zone. I don't know how much of the expensive part of DTF construction V8 commits to upon entering the constructor, though.

Conversely, if we have a string in the ZDT's [[TimeZone]] slot then we already know it's valid and canonicalized, so an implementation can optimize that part out. If we don't require that the time zone override is already valid and canonicalized when passed in to InitializeDateTimeFormat, then we lose that information, and an implementation can't skip the validation and canonicalization.

Reply:
IIRC, my concern was that if the time zone was built-in, then we should just assert that it is already canonicalized, and custom IANA time zones aren't supported anyway in DTF.